src/common/user.ts

```diff
@@ -13,34 +12,45 @@ import {
   Settings,
   SourceDisplay,
   SourceRequest,
+  UserSubscriptionStatus,
+  SubscriptionProvider,
   User,
   View,
 } from '../entity';
 import { ghostUser } from './index';
 import { cancelSubscription } from './paddle';
 import type { AuthContext } from '../Context';
 import { ForbiddenError } from 'apollo-server-errors';
+import { logger } from '../logger';
 
 export const deleteUser = async (
   con: DataSource,
-  logger: FastifyLoggerInstance,
   userId: string,
   messageId?: string,
 ) => {
   try {
-    const user = await con.getRepository(User).findOne({
+    const { subscriptionFlags } = await con.getRepository(User).findOneOrFail({
       select: ['subscriptionFlags'],
       where: { id: userId },
     });
-    if (user?.subscriptionFlags?.subscriptionId) {
+    if (subscriptionFlags?.subscriptionId) {
+      if (
+        subscriptionFlags?.provider === SubscriptionProvider.AppleStoreKit &&
+        subscriptionFlags?.status === UserSubscriptionStatus.Active
+      ) {
+        throw new Error(
+          'Apple subscriptions are not supported for user deletion',
```

Comment:
This will then block user deletion?

Reply:
How are we handling it then on product side?

Reply:
can we also add a test for this?

Reply:
Yeah, it will block user deletion. Forgot to add test.
On product side I'll disable the delete button

Reply:
With a message to tell them to cancel the subscription

Reply:
Nice, so handled on frontend/app side?

Reply:
Not yet, but I'll add it

Reply:
Tests have been added.